java/src/jmh/java/com/mlt/OmtDecoderBenchmark.java

```diff
@@ -0,0 +1,190 @@
+
+/**
+ * Benchmark for the decoding performance of OpenMapTiles schema based tiles into the MVT and MLT
+ * in-memory representations.
+ */
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.AverageTime)
+@Threads(value = 1)
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+@Fork(value = 1)
+public class OmtDecoderBenchmark {
+    private static Map<Integer, ByteArrayInputStream> encodedMvtTiles = new HashMap<>();
+    private static Map<Integer, byte[]> encodedMltTiles = new HashMap<>();
+    private static Map<Integer, MltTilesetMetadata.TileSetMetadata> tileMetadata = new HashMap<>();
+
+    @Setup
+    public void setup() throws IOException {
+        encodeTile(2, 2, 2);
+        encodeTile(4, 8, 10);
+        encodeTile(5, 16, 21);
+        encodeTile(7, 66, 84);
+        encodeTile(9, 265, 341);
+        encodeTile(10, 532, 682);
+        encodeTile(12, 2132, 2734);
+        encodeTile(14, 8298, 10748);
+    }
+
+    @Setup(Level.Invocation)
+    public void resetInputStreams() {
+        for(var is : encodedMvtTiles.values()) {
+            is.reset();
+        }
+    }
+
+    private void encodeTile(int z, int x, int y) throws IOException {
+        var encodedMvtTile = getMvtFile(z, x, y);
+        encodedMvtTiles.put(z, encodedMvtTile.getLeft());
+
+        var columnMapping = new ColumnMapping("name", ":", true);
+        var columnMappings = Optional.of(List.of(columnMapping));
+        var metadata = MltConverter.createTilesetMetadata(encodedMvtTile.getRight(), columnMappings, true);
+        tileMetadata.put(z, metadata);
+
+        var allowIdRegeneration = true;
+        var allowSorting = false;
+        var optimization = new FeatureTableOptimizations(allowSorting, allowIdRegeneration, columnMappings);
+        var optimizations = Map.of("place", optimization, "water_name", optimization, "transportation",
+                optimization, "transportation_name", optimization, "park", optimization, "mountain_peak",
+                optimization, "poi", optimization, "waterway", optimization, "aerodrome_label", optimization);
+        var encodedMltTile = MltConverter.convertMvt(encodedMvtTile.getRight(), new ConversionConfig(true, true, optimizations),
+                metadata);
+        encodedMltTiles.put(z, encodedMltTile);
+    }
+
+    private Pair<ByteArrayInputStream, MapboxVectorTile> getMvtFile(int z, int x, int y) throws IOException {
+        var tileId = String.format("%s_%s_%s", z, x, y);
+        var mvtFilePath = Paths.get(com.mlt.test.constants.TestConstants.OMT_MVT_PATH, tileId + ".mvt" );
+        var encodedTile =  Files.readAllBytes(mvtFilePath);
+        var decodedTile = MvtUtils.decodeMvt(mvtFilePath);
+        return Pair.of(new ByteArrayInputStream(encodedTile), decodedTile);
+    }
+
+    @Benchmark
+    public void decodeMvtZ2() throws IOException {
+        var mvTile = encodedMvtTiles.get(2);
+        var decodedMvTile = MvtUtils.decodeMvt2Fast(mvTile);
```

Comment:
@mactrem I don't know much about JVM optimizations, so this may not apply: but I wonder if it might be important to do something with this variable in each benchmark (like asserting on the final size or something like that) to ensure that the JVM optimizer does not do anything to throw away work (since I presume it can detect an unused variable).